rust/core-lib/src/editor.rs

```diff
@@ -901,6 +901,11 @@ impl Editor {
         }
     }
 
+    fn cancel_operation(&mut self) {
```

Comment:
nit: the convention we've been using is to preface methods that are directly handling an RPC with `do_`. 

Otherwise this looks great! I've added xi-mac support here: https://github.com/google/xi-mac/pull/84